### PR TITLE
Validate payload when creating jobs

### PR DESCRIPTION
**Pull request by rkondratowicz**
No story for this one, nor was it mentioned in any other stories, but...

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/model/HttpAction.java [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.hibernate.validator.constraints.NotBlank;
 import org.springframework.http.HttpMethod;
 
 import java.util.Map;
+import javax.validation.constraints.NotNull;
 
 public class HttpAction {
 
+    @NotBlank
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/model/HttpAction.java]: Maybe we could require a valid URL here?

Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/model/HttpAction.java:13]: Maybe we could require a valid URL here?

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/ResponseExceptionHandler.java [hunk 2]
```diff
@@ -16,6 +18,11 @@
 @ControllerAdvice
 public class ResponseExceptionHandler extends ResponseEntityExceptionHandler {
 
+    @InitBinder
+    private void activateDirectFieldAccess(DataBinder dataBinder) {
+        dataBinder.initDirectFieldAccess();
+    }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/ResponseExceptionHandler.java]: required if there are no getters/setters

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsController.java [hunk 3]
```diff
@@ -38,7 +42,7 @@ public JobsController(JobsService jobsService, AuthService authService) {
     @PostMapping(path = "")
     @ApiOperation("Create a new job")
     public ResponseEntity<Void> create(
-        @RequestBody Job job,
+        @Valid @RequestBody Job job,
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsController.java]: I know that technically it is honky dory, but having annotations in separate lines would be very appreciated. 🥇

File: src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java [hunk 4]
```diff
@@ -45,16 +46,24 @@ public void should_fill_location_header_on_successful_save() throws Exception {
...
+        send(SampleData.jobJson())
             .andExpect(header().string(LOCATION, endsWith("/jobs/" + id)));
     }
 
+    @Test
+    public void should_validate_model() throws Exception {
+        String jobWithoutUrl = SampleData.jobJson("");
+
+        send(jobWithoutUrl)
+            .andExpect(status().isBadRequest());
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java:57]: What's the response here? Is it just a bad request or is there a body indicating what's wrong? I'm thinking about covering other fields.

File: src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java [hunk 5]
```diff
@@ -45,16 +46,24 @@ public void should_fill_location_header_on_successful_save() throws Exception {
...
             .willReturn(id);
 
-        send(SampleData.validJobJson())
+        send(SampleData.jobJson())
             .andExpect(header().string(LOCATION, endsWith("/jobs/" + id)));
     }
 
+    @Test
+    public void should_validate_model() throws Exception {
+        String jobWithoutUrl = SampleData.jobJson("");
```
Comment (KarolJastrzebski) [inline src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java:54]: Would validation report also malformed URLs?

File: src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java (hunk 4 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java:57]: currently response body is empty

File: src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java (hunk 5 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/jobscheduler/controllers/jobscontroller/CreateTest.java:54]: we only test whether validation happened here

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/model/HttpAction.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/jobscheduler/model/HttpAction.java:13]: done

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsController.java (hunk 3 above)
Comment (KarolJastrzebski) [reply src/main/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsController.java]: Sorry, I just noticed it's in method... Ignore me!

**Review by lgonczar — APPROVED** [review]
My preference would be to check if validation fails for each annotated field (empty when required, not a valid url, etc.), because if we take out an annotation, we're going to miss validation, but the tests would still pass. Up to you.

**Review by KarolJastrzebski — APPROVED** [review]

**Pull request merged**